docs(react-native): UI Component - IncomingCallView (#812)
GetStream/stream-video-js: packages/react-native-sdk/docusaurus/docs/reactnative/04-ui-components/call/incoming-call-view.mdx

title: IncomingCallView
---

> IncomingCallView represents the incoming call state and the UI, when a user receives a call from someone else.

Reviewer: ```suggestion
IncomingCallView represents the incoming call state and the UI when a user receives a call from someone else.
```

  if (callingState === CallingState.RINGING && !isCallCreatedByMe) {
    return <IncomingCallView />;
  }
};

const Call = () => {
  const calls = useCalls();

  return (
    <StreamCall call={call[0]}>
      <CallPanel />
    </StreamCall>
  );
};
```

## Props

### `acceptCallButton`

Prop for the accept call button that contains all the necessary elements to make it functional.

| Type                                    |
| --------------------------------------- |
| [AcceptCallButton](#acceptcallbutton-1) |

### `rejectCallButton`

| Type                                    |
| --------------------------------------- |
| [RejectCallButton](#rejectcallbutton-1) |

Prop for the reject call button that contains all the necessary elements to make it functional.

## Button Prop

### `AcceptCallButton`

| Name             | Description                                             | Type                        |
| ---------------- | ------------------------------------------------------- | --------------------------- |
| `onPressHandler` | Callback to be called when the accept button is pressed | `() => void` \| `undefined` |

> The mostly handles the accept call button to override the default `onPress` event.

Reviewer: ```suggestion
This overrides the default `onPress` event of the accept button press.
```

### `RejectCallButton`

| Name             | Description                                             | Type                        |
| ---------------- | ------------------------------------------------------- | --------------------------- |
| `onPressHandler` | Callback to be called when the reject button is pressed | `() => void` \| `undefined` |

> The mostly handles the reject call button to override the default `onPress` event.

Reviewer: This overrides the default `onPress` event of the reject button.

> The mostly handles the reject call button to override the default `onPress` event.

Reviewer: ```suggestion
This overrides the default onPress event of the reject button.
```
